### PR TITLE
feat(ST-105): sent mail when student enroll in a class

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Installed packages: @nestjs-modules/mailer and nodemailer
- Created mail.module.ts file and added it on app.module.ts file
- Created mail.service.ts file 
- Setup the env file, hence need to change this file: environment-variables.interface.ts
- Made use of MailService in the enrollment.service layer
- Imported the MailModule in EnrollmentModule
## Associated Ticket Link(s)

- https://trello.com/c/8Va2wVkq

## Screenshots / Screen Recordings

N/A

### CODE REVIEW

File: src/common/interfaces/environment-variables.interface.ts [hunk 1]
```diff
@@ -10,4 +10,8 @@ export interface IEnvironmentVariables {
   AWS_S3_ENDPOINT: string;
   AWS_S3_BUCKET_NAME: string;
   AWS_S3_PRESIGN_URL_EXPIRY_IN_MINUTES: number;
+  MAIL_HOST: string;
+  MAIL_PORT: number;
+  MAIL_USER: string;
+  MAIL_PASSWORD: string;
```
Comment (ishmam-iqbal-sazim) [inline src/common/interfaces/environment-variables.interface.ts:16]: Could you add these to the .env.example as well?

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

File: src/common/interfaces/environment-variables.interface.ts (hunk 1 above)
Comment (ghost) [reply src/common/interfaces/environment-variables.interface.ts:16]: Okay bhaiya, will do that.

**Comment by ragib-hossain-sazim** [comment]
[Super Nitpick on PR title] `sent mail when student enroll in a class` => `Send mail to student when enrolling in a class`

File: .env.example [hunk 2]
```diff
@@ -14,3 +14,8 @@ AWS_S3_REGION="us-east-1"
 AWS_S3_ENDPOINT="http://localhost:4566"
 AWS_S3_BUCKET_NAME="project-dev-bucket"
 AWS_S3_PRESIGN_URL_EXPIRY_IN_MINUTES="5"
+
+MAIL_HOST: string;
```
Comment (ragib-hossain-sazim) [inline .env.example]: This secrets should hold sample values, not their data type.

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending comments

**Pull request merged**